pkg/ddl/ddl_worker.go

```diff
@@ -970,6 +975,12 @@ func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
 	writeBinlog(d.binlogCli, txn, job)
 	// reset the SQL digest to make topsql work right.
 	w.sess.GetSessionVars().StmtCtx.ResetSQLDigest(job.Query)
+	if !w.ddlCtx.isOwner() {
+		// This TiDB instance is not DDL owner anymore, it should not commit any transaction.
+		w.sess.Rollback()
+		d.unlockSchemaVersion(job.ID)
+		return 0, dbterror.ErrNotOwner
+	}
 	err = w.sess.Commit()
```

Comment:
Could we encapsulate the check and the commit into a function? In this way, we can more uniform processing.

Reply:
For `HandleJobDone` we cannot wrap it together with `commit` because `finishDDLJob` has other logic.

Reply:
1fc432f9a8b293040deb3b0104bad80020a09c78